test_conformance/subgroups/subgroup_common_templates.h

```diff
@@ -388,11 +390,57 @@ template <typename Ty> bool is_floating_point()
         || std::is_same<Ty, subgroups::cl_half>::value;
 }
 
+// limit possible input values to avoid arithmetic rounding/overflow issues.
+// for each subgroup values defined different values
+// for rest of workitems set 1
+// shuffle values
+template <typename Ty>
+void fill_and_shuffle_safe_values(std::vector<cl_ulong> &safe_values,
+                                  const int &sb_size)
+{
+    // below values are 0.5f, 0.25f, 2.0f, 4.0f
+    std::vector<cl_ulong> safe_half_values{ 0x3800, 0x3400, 0x4000, 0x4400 };
```

Comment:
Does this really do what we want it to do?  Since we take the incoming cl_ulong value then convert to float, then convert to half, it seems like we will generate very large half values instead of 0.5h, 0.25h, 2.0h, and 4.0h as intended:

```c++
// set scalar value to half scalar
template <typename Ty>
typename std::enable_if<TypeManager<Ty>::is_sb_scalar_type::value>::type
set_value(Ty &lhs, const cl_ulong &rhs)
{
    lhs.data = cl_half_from_float(static_cast<cl_float>(rhs), g_rounding_mode);
}
```

Reply:
I are right. Fixed in latest